Block/ProductLabel/ProductLabel.php

```diff
@@ -116,13 +183,15 @@ public function getProductLabels()
         $attributesProduct = $this->getAttributesOfCurrentProduct();
 
         foreach ($productLabelList as $productLabel) {
-            $attributeIdLabel = $productLabel['attribute_id'];
-            $optionIdLabel = $productLabel['option_id'];
-            foreach ($attributesProduct as $attribute) {
-                if (isset($attribute['id']) && ($attributeIdLabel == $attribute['id'])){
-                    if (!is_array($attribute['options'])) {
-                        if (isset($attribute['options']) && ($optionIdLabel == $attribute['options'])) {
-                            $productLabels [] = $productLabel;
+            if ($productLabel['is_active']) {
```

Comment:
This does not work well with multiple-value based attributes. Example : on sample data catalog I have attached a label to the "Climate" attribute, with "Rainy" value.

BUT products can have several values for the attribute Climate, so a strict compare with `==` is not enough.

Probably a version where we consider everything as an array of values and then test it with `in_array` would be more efficient. Something like : 

```php
foreach ($attributesProduct as $attribute) {
    if (isset($attribute['id']) && ($attributeIdLabel == $attribute['id'])) {
        $options = $attribute['options'] ?? [];
        if (!is_array($options)) {
            $options = explode(',', $options);
        }
        if (in_array($optionIdLabel, $options)) {
            $productLabels[] = $productLabel;
        }
    }
}
```

Reply:
Done